### PR TITLE
Migrate NEI-Utilities

### PR DESCRIPTION
**Pull request by slprime**
### Dotted Line
image
### Background
image
### Settings
image

## Clipboard
image

P.S.
Issue https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/15562

### CODE REVIEW

File: src/main/java/codechicken/nei/ItemHistoryPanel.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    protected int mouseDownSlot = -1;
+    protected ItemsGrid grid;
+
+    public ItemHistoryPanel() {
+        grid = new ItemsGrid();
+    }
+
+    public void draw(int mousex, int mousey) {
+
+        if (NEIClientConfig.getIntSetting("inventory.history.splittingMode") == 0) {
```
Comment (Caedis) [inline src/main/java/codechicken/nei/ItemHistoryPanel.java:27]: Can you cache these values instead of getting them every render tick?

Comment (slprime) [reply src/main/java/codechicken/nei/ItemHistoryPanel.java:27]: many NEIClientConfig calculated every tick. we need a systematic solution to this problem

image

image

Comment (Caedis) [reply src/main/java/codechicken/nei/ItemHistoryPanel.java:27]: This is fine for now, but something we should look into in the future

**Comment by Caedis** [comment]
Should add a check if NEI-Utilities is loaded and maybe throw an error to stop loading?

**Comment by Caedis** [comment]
What happens if a player has NEI-Utilities loaded as well?

**Comment by slprime** [comment]
> What happens if a player has NEI-Utilities loaded as well?

image

**Comment by Caedis** [comment]
Should prevent that

**Comment by slprime** [comment]
NEI cannot prevent this because nei-utilities patches ItemPanel

**Review by Caedis — APPROVED** [review]

**Pull request merged**